lib/promscrape/config.go

```diff
@@ -1144,6 +1144,8 @@ func (sdc *FileSDConfig) appendScrapeWork(dst []*ScrapeWork, swsMapPrev map[stri
 				if sws := swsMapPrev[path]; sws != nil {
 					// Re-use the previous valid scrape work for this path.
 					logger.Errorf("keeping the previously loaded `static_configs` from %q because of error when re-loading the file: %s", path, err)
+					// Remove all scrape work because all of them will be added from the previous scrape work map.
+					dst = dst[:0]
```

Comment:
I don't understand this change...

dst is created from scratch each time `appendScrapeWork` is called. How this explains the behavior you described in the commit message?

Also, if you reset the slice here, then all previously parsed work will be lost. So if you have 10 files to parse and there is an err on 6th file, then in result you'll get only 3 files because previously parsed 5 files will be reset. Is my understanding correct?

Reply:
Yes, you are right, I found the problem. Will implement another solution